source/common/http/header_map_impl.cc

```diff
@@ -362,8 +362,10 @@ void HeaderMapImpl::insertByKey(HeaderString&& key, HeaderString&& value) {
     if (*lookup.value().entry_ == nullptr) {
       maybeCreateInline(lookup.value().entry_, *lookup.value().key_, std::move(value));
     } else {
+      const std::string delimiter =
+          (*lookup.value().key_ == Http::Headers::get().Cookie ? "; " : ",");
```

Comment:
What does Envoy with cookie headers in the case of HTTP2?  Does this change alter HTTP1 and HTTP2 behavior?

Would be good to enhance coverage for multi-valued set-cookie and cookie headers in test/integration

Reply:
I suspect this is a good change, but can be done without default inlining and isn't QUIC-specific.  Maybe break out and add relnote and runtime guards?